content/flux/v0.x/stdlib/influxdata/influxdb/from.md

```diff
@@ -90,11 +90,48 @@ If authentication is _disabled_, provide an empty string (`""`).
 If authentication is _enabled_, provide your InfluxDB username and password
 using the `<username>:<password>` syntax.
 
+## Push down optimizations
+
+Some transformations after the call to `from()` will trigger performance optimizations called push downs.
+These optimizations are "pushed down" from Flux into the storage layer and involve utilizing some code in storage to directly apply the transformation to the data to either filter or aggregate the data.
+These happen automatically and the person writing the query doesn't have to know about the technical details, but it is helpful to know about these optimizations so you can understand why certain orders of transformations may be faster than others.
+
+Push downs require the chain between transformations to be unbroken and exclusive.
+A `from()` that goes to multiple push down optimizations will cause none of them to be applied.
+To avoid this while allowing for code reuse, prefer invoking `from()` in a function when multiple push downs get applied.
+
+### Filter
+
+A `filter()` that makes a comparison to `r._measurement`, `r._field`, `r._value` or any tag value will be pushed down to the storage layer.
+Comparisons that use functions will not.
+If the function produces a static value, you can evaluate the function outside of the filter and the result will be pushed down.
+Multiple consecutive filters that would get pushed down will get merged together into a single filter that gets pushed down.
```

Comment:
~~~suggestion
`filter()` transformations that compare `r._measurement`, `r._field`, `r._value` or any tag value are pushed down to the storage layer.
Comparisons that use functions do not.
If the function produces a static value, evaluate the function outside of `filter()`.
For example:

```js\
import "strings"

// filter() is NOT pushed down
data
    |> filter(fn: (r) => r.example == strings.joinStr(arr: ["foo", "bar"], v: ""))

// filter() is pushed down
exVar = strings.joinStr(arr: ["foo", "bar"], v: ""))

data
    |> filter(fn: (r) => r.example == exVar)
```

Multiple consecutive `filter()` transformations that can be pushed down are merged together into a single filter that gets pushed down.
~~~

Reply:
This is also true for string concatenation and interpolation, right? Filters that concat or interpolate strings can't be pushed down.

Reply:
I think it probably applies to interpolation and concatenation but honestly it probably shouldn't.